src/Story.php

```diff
@@ -29,40 +32,124 @@ final public static function load(): self
     }
 
     /**
+     * Get a random item from a pool.
+     */
+    final public static function getRandom(string $pool): Proxy
+    {
+        return static::getRandomSet($pool, 1)[0];
+    }
+
+    /**
+     * Get a random set of items from a pool.
+     *
+     * @return Proxy[]
+     */
+    final public static function getRandomSet(string $pool, int $number): array
+    {
+        if ($number < 0) {
+            throw new \InvalidArgumentException(\sprintf('$number must be positive (%d given).', $number));
+        }
+
+        return static::getRandomRange($pool, $number, $number);
+    }
+
+    /**
+     * Get a random range of items from a pool.
+     *
+     * @return Proxy[]
+     */
+    final public static function getRandomRange(string $pool, int $min, int $max): array
+    {
+        if ($min < 0) {
+            throw new \InvalidArgumentException(\sprintf('$min must be positive (%d given).', $min));
+        }
+
+        if ($max < $min) {
+            throw new \InvalidArgumentException(\sprintf('$max (%d) cannot be less than $min (%d).', $max, $min));
+        }
+
+        $story = static::load();
+        $values = $story->pools[$pool] ?? [];
+
+        \shuffle($values);
+
+        if (\count($values) < $max) {
+            throw new \RuntimeException(\sprintf('At least %d items must be on pool "%s" (%d items found).', $max, $pool, \count($values)));
+        }
+
+        return \array_slice($values, 0, \random_int($min, $max));
+    }
+
+    /**
+     * @param object|Proxy|Factory $object
+     *
      * @return static
      */
-    final public function add(string $name, $state): self
+    final public function add(string $name, object $object, string ...$pools): self
```

Comment:
Should I make this `string|array $pools` (we've previously chosen to avoid variadic variables)?

Reply:
Wouldn't it be better to force the array type instead? And also have `addToPools()` in addition to `addToPool()`

Variadic arguments are an issue if you want to add more args after that.

Reply:
Adding it to multiple pools seems very advanced to me. I would suggest only implementing `string $pool` now and then accepting a community contribution if someone really needs to add one fixture to multiple pools.

Reply:
Ok, I've simplified - the PR description is updated to the new API.